components/Navbar/NavbarDesktop/Tab.tsx

```diff
@@ -39,14 +39,33 @@ export default function Tab({
   const isRouteLesson = pathData.length === 4
 
   const { progress } = useProgressContext()
-  const { isUnlocked, isCompleted } = useLessonStatus(
+  const { isUnlocked } = useLessonStatus(
+    progress,
+    getLessonKey(
+      slug,
+      challenge.lessonId === chapters[slug].metadata.lessons[0]
+        ? 'intro-1'
+        : challenge.lessonId
+    )
+  )
+  const { isCompleted } = useLessonStatus(
     progress,
     getLessonKey(slug, challenge.lessonId)
   )
 
-  const challengeId = isRouteLesson ? pathData.pop().split('-')[0] : undefined
+  const challengeId = isRouteLesson
+    ? pathData
+        .pop()
+        .split('-')[0]
+        .replace('intro', chapters[slug].metadata.challenges[0].split('-')[0])
```

Comment:
What is happening here? It seems to take intro-1, and replaces `intro` with the chapter's first challenge lessonId so it would become something like `scripting-1`?

Reply:
This allows the intro pages to be treated as challenges by the isActive variable, while not moving the intro pages to be a part of a specific lesson

Reply:
Okay, feels a bit hacky but its fine, we can always clean up later.